cpp/include/cudf/io/types.hpp

```diff
@@ -169,6 +171,25 @@ struct host_buffer {
   host_buffer(const char* data, size_t size) : data(data), size(size) {}
 };
 
+/**
+ * @brief Non-owning view of a device memory buffer
+ *
+ * Used to describe buffer input in `source_info` objects.
+ */
+// TODO: to be replaced by `device_span`?
```

Comment:
This was precisely what I was going to ask. Why not just use `device_span<byte>`?

Reply:
Use `device_span` instead.